### ISSUE TITLE
plugin boilerplate or yo generator

### ISSUE THREAD

**Issue by garrettjstevens** [GMOD/jbrowse-components#22]
@rbuels I thought it would be nice to have a sample run-time plugin that does something basic that we could use to experiment with. I could work on this if you want, but I'd need a bit of direction since I'm not entirely sure how we are going to implement them.

**Comment by rbuels** [comment]
It would be a good idea to look at what Atom, vscode, and other plugin-based javascript systems do for their plugins.

**Comment by garrettjstevens** [comment]
Not just for runtime plugins, but for plugins in general, I really like the idea of having a Yeoman generator for plugins like VS Code [has for their extensions](https://code.visualstudio.com/docs/extensions/yocode). If we could have several different hello-world-style templates that create plugins with different combinations of pluggable elements, I think it would go a long way toward making plugin creation easier and more consistent.

**Comment by rbuels** [comment]
Yes a yo generator and/or some boilerplate projects would be great

**Comment by cmdcolin** [comment]
Just a random thing is that currently plugin loading errors give an uncaught exception. I decided to not make a separate issue as this code will receive changes when we address this issue

**Comment by elliothershberg** [comment]
I think it's worth noting @garrettjstevens and I have discussed this idea several times when thinking about the plugin template and #1146 . While the template is fairly automated now and includes an example, it could still be neat to have a generator tool for making plugins instead of a template. I think the value would be in having sub-generators to scaffold the boilerplate for various types of widgets.

**Comment by cmdcolin** [comment]
Personally I'm a bigger fan of having working example plugins rather than generators, and people can base their work off examples that we have

**Comment by garrettjstevens** [comment]
We have a boilerplate template here: https://github.com/GMOD/jbrowse-plugin-template. We may open another issue later to do a generator for plugins.

**Comment by elliothershberg** [comment]
@rbuels and I were talking about the prospect of revisiting generators for plugins. One potential implementation to base ours off of could be the [RedwoodJS](https://redwoodjs.com/) generators. They rolled their own basically, and didn't use a scaffolding tool. 

Their generator code is here: [https://github.com/redwoodjs/redwood/tree/main/packages/internal/src/generate](https://github.com/redwoodjs/redwood/tree/main/packages/internal/src/generate)

I've found generators to lead to a really nice developer experience in other cases where I was using tools that had them, and felt it greatly reduced the learning code. I realize that might not be the majority sentiment here, but I thought I'd re-open another time to discuss 😄

**Comment by cmdcolin** [comment]
i'd advocate, if needed, to use the "create-react-app" model of generating a boilerplate
